backend/services/implementations/__tests__/classService.test.ts

```diff
@@ -106,6 +106,35 @@ describe("mongo classService", (): void => {
     );
   });
 
+  /* uncomment after adding logic to attach test sessions to class objects
+  it("getClassByTestSessionId for valid testSessionId", async () => {
+    const savedClass = await ClassModel.create(testClass[0]);
+    const res = await classService.getClassByTestSessionId(
+      savedClass.testSessions[0],
+    );
+    assertResponseMatchesExpected(savedClass, res);
+  });
+
+  it("getClassByTestSessionId for non-existing testSessionId", async () => {
+    const notFoundId = "86cb91bdc3464f14678934cd";
+    await expect(async () => {
+      await classService.getClassByTestSessionId(notFoundId);
+    }).rejects.toThrowError(
+      `Class with test session id ${notFoundId} not found`,
+    );
+  });
+
+  it("getClassByTestSessionId for classes with same testSessionId", async () => {
+    await ClassModel.create(testClass[0]);
+    await ClassModel.create(testClass[0]);
+    const testSessionId = testClass[0].testSessions[0];
+    await expect(async () => {
+      await classService.getClassByTestSessionId(testSessionId);
+    }).rejects.toThrowError(
+      `More than one class has the same Test Session of id ${testSessionId}`,
+    );
+  }); */
+
```

Comment:
originally, i had created these tests before david removed adding test sessions from the `createClass` logic. but now that the logic for adding test sessions to classes doesn't exist yet, i don't think there's a way to write unit tests for the `getClassByTestSessionId` function and i think adding that logic is beyond the scope of this PR!

so leaving tests off of this pr and  created a separate [ticket](https://www.notion.so/uwblueprintexecs/Update-backend-to-support-distributing-assessment-7efdcc435f2c46a097cb678839139a07?pvs=4) to track that work. I also left a note to uncomment these tests (and get them working) after this functionality is added!

Reply:
wait i'm confused - can we use [testClassWithTestSessions](https://github.com/uwblueprint/jump-math/blob/staging/backend/testUtils/class.ts#L74) for these tests? or is there something else missing

Reply:
wait i'm not sure if i'm misunderstanding, b/c i see one test case that uses `testClassWithTestSessions`. but didn't we remove the logic to include test sessions in `createClass`? so, how would do we store the `testSession` in the RequestDTO?

Reply:
the `createClass` we wrote doesn't support adding test sessions but in the unit test you can use Model.create() to still create an object with test sessions (to mimick having called createClass + addTestSessionToClass). not sure if that answers your question

Reply:
the test case that uses it is a getter, so we would expect some objects in our database to already have test sessions